.circleci/config.yml

```diff
@@ -158,38 +143,17 @@ jobs:
       - cleanup-docker
 
   # Try to build s2i builder image and run smoke tests
-  s2i-builder:
-    executor: s2i
+  build-development-image:
+    executor: docker
     environment:
-      IMAGE_NAME: apicast-builder-image
+      DEVEL_IMAGE: apicast-development
     steps:
-      - setup-s2i
+      - setup-build-env
       - run:
-          name: "Build image"
+          name: "Build development image"
           command: |
-            docker rmi --no-prune ${IMAGE_NAME} 2>/dev/null || :
-            make builder-image
-
-      - run:
-          name: "Build image (incremental)"
-          command: make builder-image
-
-      - run: |
-          make test-builder-image gateway-logs --keep-going
-          make lint-schema
-
-      - login-docker:
-          command: |
-            CIRCLE_BRANCH_LOWER=`echo $CIRCLE_BRANCH | tr '[:upper:]' '[:lower:]'`
-            IMAGE_TAG="${CIRCLE_TAG:-${CIRCLE_BRANCH_LOWER}}-builder"
-            if [ "${IMAGE_TAG}" == "master-builder" ]
-            then
-              IMAGE_TAG="latest-builder"
-            fi
-            IMAGE_NAME="apicast:${IMAGE_TAG}"
-            docker tag apicast-builder-image ${IMAGE_NAME}
-            make push
-
+            docker rmi --no-prune ${DEVEL_IMAGE} 2>/dev/null || :
+            make dev-build
```

Comment:
I would expect us to be executing the make target which then runs those smoke tests no? Probably rename the make target `test-runtime-image` to `test-dev-image` and execute that. Would need to pass in the correct image which looks to be `quay.io/3scale/apicast-ci:openresty-1.19.3`. However, given that is a fixed tag rather than floating we might want to think about a better way to reference that otherwise we would need to update this circleci config every time the image updates??

Reply:
The thing is that the development image is a ubi8 + openresty. There is no apicast in there at all. So what to tests?

I have checked the former devel image repo, [s2i-openresty](https://github.com/3scale/s2i-openresty). There is a makefile target for tests. What are those test about? The summary is on the header of [test/run](https://github.com/3scale/s2i-openresty/blob/master/test/run) file:
```
# The 'run' performs a simple test that verifies the S2I image.
# The main focus here is to exercise the S2I scripts.
#
# For more information see the documentation:
# https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md
```
The devel image of s2i-openresty has scripts that are run in the s2i workflow at different stages. The tests are meant to test the scripts work as expected.

However, the current base image is nothing else than a set of packages installed, some env variables set and so on. So, even though we could come up with some tests, I do not think it is worth doing. wdyt?

Reply:
The development image does not have any apicast code. 

The development container (running the development image) has mounted your `$PWD` in `/opt/app-root/src`. Configured [here](https://github.com/3scale/APIcast/blob/remove-s2i/docker-compose-devel-volmount-default.yml). With this approach, you change the apicast lua code or whatever using your favorite IDE and the changes are immediately available in the development container.

Reply:
alright that makes a lot of sense now thanks for clarifying @eguzki. Then I guess I would agree with your statement above: probably it's not necessary to test this image.